freqtrade/plugins/protections/iprotection.py

```diff
@@ -33,21 +33,33 @@ def __init__(self, config: Config, protection_config: Dict[str, Any]) -> None:
         self._protection_config = protection_config
         self._stop_duration_candles: Optional[int] = None
         self._lookback_period_candles: Optional[int] = None
+        self.unlock_at: Optional[datetime] = None
 
         tf_in_min = timeframe_to_minutes(config["timeframe"])
         if "stop_duration_candles" in protection_config:
             self._stop_duration_candles = int(protection_config.get("stop_duration_candles", 1))
             self._stop_duration = tf_in_min * self._stop_duration_candles
         else:
-            self._stop_duration_candles = None
             self._stop_duration = int(protection_config.get("stop_duration", 60))
         if "lookback_period_candles" in protection_config:
             self._lookback_period_candles = int(protection_config.get("lookback_period_candles", 1))
             self._lookback_period = tf_in_min * self._lookback_period_candles
         else:
-            self._lookback_period_candles = None
             self._lookback_period = int(protection_config.get("lookback_period", 60))
 
+        if "unlock_at" in protection_config:
+            now_time = datetime.now(timezone.utc)
+            unlock_at = datetime.strptime(protection_config["unlock_at"], "%H:%M").replace(
+                day=now_time.day, year=now_time.year, month=now_time.month
+            )
+
+            if unlock_at.time() < now_time.time():
+                unlock_at = unlock_at.replace(day=now_time.day + 1)
+
+            unlock_at = unlock_at.replace(tzinfo=timezone.utc)
+            self._stop_duration = self.calculate_timespan(now_time, unlock_at)
```

Comment:
This happens on init (not when the protection runs)... why would you move the day to tomorrow? 
That also makes the "stop_duration" calculation kinda odd, no?

what if the protection triggers the day after tomorrow?

Reply:
Ah okay, I moved the day to tomorrow in order to calculate a stop duration to a time in the future when the time is already met on the current day.

Reply:
well yes - but not here :laughing: 
you'll need that logic - sure - but when the protection triggers - NOT when the bot is started ;)

Reply:
How about adding it to global_stop instead?

Reply:
more like `calculate_lock_end()` ? 

though i didn't look at how to "fix" it in detail.

Reply:
Good idea, tried to did it that way.